kernel/src/main/java/org/kframework/lsp/KTextDocumentService.java

```diff
@@ -0,0 +1,194 @@
+                                                        completionItem.setDetail("module " + m.getName());
+                                                        String doc = "syntax ";
+                                                        doc += !s.getParams().isEmpty() ?
+                                                                "{" + s.getParams().stream().map(Sort::toString).collect(Collectors.joining(", ")) + "} " : "";
+                                                        doc += s.getDeclaredSort() + " ::= ";
+                                                        doc += p.toString();
+                                                        completionItem.setDocumentation(doc);
+                                                        completionItem.setKind(CompletionItemKind.Snippet);
+                                                        lci.add(completionItem);
+                                                    }
+                                                }))))));
+
+        return lci;
+    }
+
+    private List<Diagnostic> outerParse(String uri) {
+        List<DefinitionItem> di = null;
+        List<Diagnostic> problems = new ArrayList<>();
+        try {
+            String contents = FileUtil.load(new File(new URI(uri)));
+            if (uri.endsWith(".md"))
+                contents = mdExtractor.extract(contents, Source.apply(uri));
+            di = Outer.parse(Source.apply(uri), contents, null);
+            elems.put(uri, di);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        } catch (KEMException e) {
+            Location loc = e.exception.getLocation();
+            if (loc == null) loc = new Location(1, 1, 1, 2);
+            Range range = new Range(new Position(loc.startLine() - 1, loc.startColumn() - 1),
+                    new Position(loc.endLine() - 1, loc.endColumn() - 1));
+            Diagnostic d = new Diagnostic(range, e.exception.getMessage(), DiagnosticSeverity.Error, "Outer Parser");
+            problems.add(d);
+        }
+        if (problems.isEmpty())
+            this.clientLogger.logMessage("Operation 'outerParse' #di: " + di.size());
+        else
+            this.clientLogger.logMessage("Operation 'outerParse' #problems: " + problems.size());
+        return problems;
+    }
+
+    // for quick testing
+    public static void main(String[] args) throws InterruptedException, ExecutionException, URISyntaxException {
+        String uri = "file:///home/radu/work/test/test.k";
+        List<DefinitionItem> dis = Outer.parse(Source.apply(uri), FileUtil.load(new File(new URI(uri))), null);
+        List<DefinitionItem> doma = Outer.parse(Source.apply(domains.toString()), FileUtil.load(new File(domains)), null);
+        List<DefinitionItem> kst = Outer.parse(Source.apply(kast.toString()), FileUtil.load(new File(kast)), null);
+        List<CompletionItem> x = getCompletionItems(dis);
+        System.out.println(x.size());
+    }
```

Comment:
Does this need to be removed?

Reply:
I would like to keep this for now.
It makes it really easy to quickly fire up a test instead of trying to debug code through the editor.
We already have quite a few of these in the repo.

Reply:
Sure, if it helps to test then keep it - maybe take the path as a command line parameter rather than hard-coding a path from your own machine.